spring-integration-mail/src/main/java/org/springframework/integration/mail/ImapIdleChannelAdapter.java

```diff
@@ -104,20 +102,19 @@ public void setAdviceChain(List<Advice> adviceChain) {
 	}
 
 	/**
-	 * Specify an {@link Executor} used to send messages received by the
-	 * adapter.
+	 * Specify an {@link Executor} used to send messages received by the adapter.
 	 * @param sendingTaskExecutor the sendingTaskExecutor to set
+	 * @deprecated since 6.1 in favor of async hands-off downstream in the flow,
+	 * e.g. {@link  org.springframework.integration.channel.ExecutorChannel}.
 	 */
+	@Deprecated(since = "6.1", forRemoval = true)
```

Comment:
Don't we need to deprecate in 6.0 and just remove here?

Reply:
I'd love to, but that is a behavior change and making it deprecated there would just lead to always use a default one.
There is a simple workaround with setting a `SyncTaskExecutor` instead.
Plus it is not too far from `6.2` this Fall where we indeed will remove it already.